README


This archive contains the distribution YAML-Tiny,
version 1.64:

  Read/Write YAML files with as little code as possible

This software is copyright (c) 2006 by Adam Kennedy.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for YAML-Tiny

1.64      2014-10-08 02:56:30Z

        [CHANGED]

        - remove silencing of any errors encountered while loading
          Scalar::Util (GH#33, Graham Knop)
        - now using JSON::MaybeXS in tests instead of JSON.pm

1.63      2014-06-11 21:58:18Z

        [FIXED]

        - incorrect error messages fixed, when $@ is clobbered when Carp
          wasn't loaded (GH#30, GH#31, Hilko Bengen)

        [CHANGED]

        - now checking for, and erroring on, duplicate keys found in a hash
          (GH#32, Hilko Bengen)

1.62      2014-03-16 12:28:44Z

        [FIXED]

        - fix handling of trailing colon in key name (RT#92916, H.Merijn
          Brand)

1.61      2014-02-24 16:59:49Z

        [FIXED]

        - fixed a test for VMS (RT#93297, Craig Berry)

1.60      2014-02-13 20:31:56Z

        - shipping 1.59 as stable, with no changes.

1.59      2014-02-06 03:10:35Z (TRIAL RELEASE)

        [CHANGED]

        - numeric values are now quoted whenever they've been used as a string
          (fixes inconsistent behaviour seen with numeric values, due to
          differences between the XS and pure-perl variants of Data::Dumper).
          (github issue #24)
        - numeric hash keys are now always quoted.

1.58      2014-02-04 18:01:58Z

        [INCOMPATIBLE CHANGE]

        - 1.57 omitted a change entry for the following change:

        - Previously, YAML::Tiny was sloppy about file encodings.  It is
          now strict.  The 'read' method and 'LoadFile' function expect
          UTF-8 encoded files.  The 'write' method and 'DumpFile' function
          produce UTF-8 encoded files.  The 'read_string' and
          'write_string' methods and the 'Load' and 'Dump' functions
          expect or generate (decoded) character data.

1.57      2014-01-30 22:12:38Z

        [INCOMPATIBLE CHANGE]

          - Previously, some errors would throw exceptions and some would
            return the error condition in $YAML::Tiny::errstr.  Now all
            errors throw exceptions.  Use of $errstr and the errstr method
            are deprecated. (David Golden)

        [FIXED]

          - Fixed write method to encode YAML file with UTF-8 (David Golden)
          - Improved SYNOPSIS and documentation of new (David Golden)

        [TESTING]

          - Tests have been cleaned up and reorganized.  Test coverage
            has been significnatly improved. (Ingy döt Net, David Golden,
            Jim Keenan, Karen Etheridge)

1.56      2013-09-25 02:38:19Z
          - read_string documentation error fixed (RT#74409, thanks Tim Heaney!)
          - re-release with fixed compile test

1.55      2013-09-19 04:07:32Z
          - again packaging with ExtUtils::MakeMaker

1.54      2013-08-22 03:55:41Z
          - convert to Dist::Zilla
          - Updated format to conform to CPAN::Changes::Spec

1.53      2013-08-20
          - really fixed metadata

1.52      2013-08-20
          - updated repository metadata to reflect move to github

1.51      2012-03-11
          - The fix described in 1.48 was incorrect. Removing an additional
            if $@ to fix it properly.
          - Make the Scalar::Util version check resistant to bugs caused by
            development versions.

1.50      2011-06-23
          - Major bug fix, all code that writes arbitrary data should upgrade.
          - Simple scalars with no whitespace but ending in a colon like ABC: were
            not being quoted, which results in the parser confusing it with a
            mapping key and crashing.

1.49      2011-03-08
          - No functional changes.
          - Don't depend on the YAML modules in RELEASE_TESTING, as it can
            pollute the advisory META.yml.

1.48      2011-02-01
          - Fix to the refaddr compatibility where Scalar::Util is installed
            but is older than 1.18.

1.47      2011-01-31
          - No functional changes
          - Only depend on the YAML implementations when we are release testing

1.46      2010-12-16
          - No functional changes
          - Moving to a production release now CPAN Testers is green again

1.45_02   2010-12-07
          - Adding experimental support for exception->errstr conversion
          - Updating test suite yaml_error to not expect exceptions

1.45_01   2010-12-07
          - Added support for trailing line comments (INGY)
          - Added checks for some characters that are illegal or reserved
            in plain scalars (INGY)
          - Minor cleaning up of some out of date POD (ADAMK)
          - Updated AUTOMATED_TESTING dependencies to new versions (ADAMK)

1.44      2010-08-08
          - No functional changes, upgrading is not required
          - Minor tweaks to Makefile.PL
          - Minor cleanup around the refaddr emulation code

1.43      2010-07-10
          - No functional changes, upgrading is not required
          - Don't import Carp 'croak' any more, saving a few K

1.42      2010-06-25
          - No functional changes, upgrading is not required
          - Clarified documation to explain that YAML::Tiny escapes "bool"
            keywords when it writes them as strings so that YAML parsers in
            other languages won't get confused, but this does not mean that
            YAML::Tiny actually supports boolean types itself.
          - Added an extra test case for boolean keyword escaping.
          - YAML::Perl comparison tests were loading the file back in with
            YAML.pm by mistake. Corrected this and added an additional case
            that YAML::Perl doesn't support that I wasn't catching.
          - Author tests updated and moved into xt/
          - Updated the versions of the other YAML parsers we test ourselves
            against to the most recent versions of each.

1.41      2009-12-11
          - Fixes a performance bug reported by Apocalypse.
            Single line double-quoted strings longer than 32,000ish characters

Makefile.PL
# This Makefile.PL for YAML-Tiny was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.013
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.28.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.038',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.008001;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Read/Write YAML files with as little code as possible",
  "AUTHOR" => "Adam Kennedy <adamk\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.038"
  },
  "DISTNAME" => "YAML-Tiny",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "YAML::Tiny",
  "PREREQ_PM" => {
    "B" => 0,
    "Carp" => 0,
    "Exporter" => 0,
    "Fcntl" => 0,
    "Scalar::Util" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Basename" => 0,
    "File::Find" => 0,
    "File::Spec" => "0.80",
    "File::Spec::Functions" => 0,
    "File::Temp" => 0,
    "Getopt::Long" => 0,
    "IO::Dir" => 0,
    "JSON::PP" => 0,
    "List::Util" => 0,
    "Test::More" => "0.99",
    "lib" => 0,
    "utf8" => 0,
    "vars" => 0
  },
  "VERSION" => "1.64",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "B" => 0,
  "Carp" => 0,
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => 0,
  "Fcntl" => 0,
  "File::Basename" => 0,
  "File::Find" => 0,
  "File::Spec" => "0.80",
  "File::Spec::Functions" => 0,
  "File::Temp" => 0,
  "Getopt::Long" => 0,
  "IO::Dir" => 0,
  "JSON::PP" => 0,
  "List::Util" => 0,
  "Scalar::Util" => 0,
  "Test::More" => "0.99",
  "lib" => 0,
  "strict" => 0,
  "utf8" => 0,
  "vars" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for YAML-Tiny was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.009
use strict;
use warnings;

if (eval 'use Module::Build::Tiny 0.038; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for YAML-Tiny was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.007.
    use strict;
    use warnings;

    use 5.008001;
    # use Module::Build::Tiny 0.038;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.038"
      },
      "dist_abstract" => "Read/Write YAML files with as little code as possible",
      "dist_author" => [
        "Adam Kennedy <adamk\@cpan.org>"
      ],
      "dist_name" => "YAML-Tiny",
      "dist_version" => "1.64",
      "license" => "perl",
      "module_name" => "YAML::Tiny",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "B" => 0,
        "Carp" => 0,
        "Exporter" => 0,
        "Fcntl" => 0,
        "Scalar::Util" => 0,
        "perl" => "5.008001",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Basename" => 0,
        "File::Find" => 0,
        "File::Spec" => "0.80",
        "File::Spec::Functions" => 0,
        "File::Temp" => 0,
        "Getopt::Long" => 0,
        "IO::Dir" => 0,
        "JSON::PP" => 0,
        "List::Util" => 0,
        "Test::More" => "0.99",
        "lib" => 0,
        "utf8" => 0,
        "vars" => 0
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Basename" => 0,
      "File::Find" => 0,
      "File::Spec" => "0.80",
      "File::Spec::Functions" => 0,
      "File::Temp" => 0,
      "Getopt::Long" => 0,
      "IO::Dir" => 0,
      "JSON::PP" => 0,
      "List::Util" => 0,
      "Test::More" => "0.99",
      "lib" => 0,
      "utf8" => 0,
      "vars" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
